### DISCUSSION TITLE
Switching to a beatmap with identical track data does not reset position

### DISCUSSION THREAD

**Discussion by HelloYeew** [ppy/osu#19370] [Ideas]
### Type

Game behaviour

### Bug description

In song selection page, I notice on some of beatmaps that has the same song. If select these beatmap, the track will not seek to the preview time that set in beatmap file.

Here is the list of some beatmap that can reproduce :

- Mopemope : [hypercute](https://osu.ppy.sh/beatmapsets/1174616#osu/2550126) and [Hey lululu](https://osu.ppy.sh/beatmapsets/1097600#osu/2293451) beatmapset
- Paradisus-Paradoxum : [StarrStyx](https://osu.ppy.sh/beatmapsets/477640#osu/1021846), [wajinshu](https://osu.ppy.sh/beatmapsets/514181#osu/1092363), [Yasajia 714](https://osu.ppy.sh/beatmapsets/483503#osu/1072335) and [toybot](https://osu.ppy.sh/beatmapsets/485788#osu/1035897) beatmapset
- Road of Resistance : [Sotarks](https://osu.ppy.sh/beatmapsets/1109083#osu/2317735), [Monstrata](https://osu.ppy.sh/beatmapsets/399372#osu/869222) and [Kroytz](https://osu.ppy.sh/beatmapsets/567324#osu/1201636) beatmapset

So I download both Mopemope and see that it use the same audio file same as Road of resistance that audio that not has the same file name but inside the audio file is same.

![image](https://user-images.githubusercontent.com/68165621/180617156-696131c7-c5d8-4d7b-b766-0f69e2182e0e.png)

It's easy to reproduce as in screen record below. But if select these beatmaps inside `now playing` menu the track will seek to the preview time in beatmap file as expected (as I select in 4 Paradisus-Paradoxum record below)

### Screenshots or videos

Mopemope set reproduce :

https://user-images.githubusercontent.com/68165621/180616598-a1c0f0f1-969d-4474-8cfe-90eef812bab4.mp4

4 Paradisus-Paradoxum :

https://user-images.githubusercontent.com/68165621/180616662-a293c496-88b7-4ac4-b062-530fa3a290e5.mp4

### Version

2022.723.0-lazer

### Logs

[logs.zip](https://github.com/ppy/osu/files/9174338/logs.zip)

**Comment by bdach** [comment]
There is a discrepancy because the song selection screen and the now playing overlay are using different code paths for changing tracks. That said I don't really have a strong opinion which behaviour is preferable here and it's a very low impact issue.

**Comment by frenzibyte** [comment]
I think it's a regression that the track is being shared across different beatmapsets, likely to have regressed by https://github.com/ppy/osu/pull/19105 which no longer checks on the difference between `BeatmapSet` hashes.

Would even be more correct to use `BeatmapSetInfo.Equals` rather than checking hash.

**Comment by smoogipoo** [comment]
I've made this into a discussion for now because I believe this is "intended" behaviour, and I've heard whisperings on Twitter where people thought this was a cool feature. It comes as a result of the files' contents being identical.

I personally don't mind it staying as it is, but I don't know if this breaks user expectations.

**Reply by peppy** [reply]
Yeah, I'm not sure. I *think* it probably should be fixed though (when changing beatmap set). My reasoning is that the preview point can potentially be set differently for different beatmaps. ~~and tatoe~~

**Reply by EnergoStalin** [reply]
Personally i like this bug(feature). But yeah, it's disrespect preview point of map so i think mappers should decide if they okay with that or not. Maybe do a poll for them along with regular players like how you think as a mapper and as a player or both at the same time) if this not discussed already on discord's live stage. For now it still works like that so i'm happy with that. Just don't want it magically disappear one beautiful day. Didn't see final conclusion here so it was my opinion

**Reply by Reletr** [reply]
Just noticed this myself w/ a few beatmaps ([1](https://osu.ppy.sh/beatmapsets/897723#osu/1909530), [2](https://osu.ppy.sh/beatmapsets/1282804#osu/2664264), [3](https://osu.ppy.sh/beatmapsets/2564338#osu/5702137)). I personally don't like it for the same reason EnergoStalin mentioned, and it makes distinguishing between mapsets just a bit harder (albeit that's a pretty small downside). Also agree that some sort of poll should be done to get a general consensus.
